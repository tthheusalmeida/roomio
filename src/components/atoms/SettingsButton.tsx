import { IoIosArrowForward } from "react-icons/io";
import Button from "./Button";
import { IconType } from "react-icons";

interface SettingsButtonProps {
  onClick: Function;
  label: string;
  icon: IconType;
}

export default function SettingsButton({
  onClick,
  label,
  icon: Icon,
}: SettingsButtonProps) {
  return (
    <Button variant="secondary" onClick={() => onClick()}>
      <div className="flex gap-4 items-center min-w-[50dvw]">
        <Icon size={24} />
        {label}
      </div>
    </Button>
  );
}
