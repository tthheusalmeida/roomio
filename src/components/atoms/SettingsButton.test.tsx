import { render, screen, fireEvent } from "@testing-library/react";
import SettingsButton from "./SettingsButton";
import { IoIosArrowForward } from "react-icons/io";

jest.mock("./Button", () => ({
  __esModule: true,
  default: ({ children, onClick }: any) => (
    <button onClick={onClick}>{children}</button>
  ),
}));

jest.mock("react-icons/io", () => ({
  IoIosArrowForward: (props: any) => <span data-testid="icon" {...props} />,
}));

describe("SettingsButton component", () => {
  it("renders the icon and label", () => {
    const { asFragment } = render(
      <SettingsButton
        onClick={() => {}}
        label="Next"
        icon={IoIosArrowForward}
      />
    );

    expect(screen.getByText("Next")).toBeInTheDocument();
    expect(screen.getByTestId("icon")).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it("calls onClick when clicked", () => {
    const handleClick = jest.fn();
    render(
      <SettingsButton
        onClick={handleClick}
        label="Next"
        icon={IoIosArrowForward}
      />
    );

    const button = screen.getByText("Next").closest("button")!;
    fireEvent.click(button);

    expect(handleClick).toHaveBeenCalled();
  });
});
