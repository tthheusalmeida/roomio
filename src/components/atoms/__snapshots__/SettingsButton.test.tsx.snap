// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SettingsButton component renders the icon and label 1`] = `
<DocumentFragment>
  <button>
    <div
      class="flex gap-4 items-center min-w-[50dvw]"
    >
      <span
        data-testid="icon"
        size="24"
      />
      Next
    </div>
  </button>
</DocumentFragment>
`;
